src/c++/perf_analyzer/idle_timer.h

```diff
@@ -0,0 +1,87 @@
+// Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+#include <chrono>
+
+namespace triton { namespace perfanalyzer {
+
+#ifndef DOCTEST_CONFIG_DISABLE
+class TestLoadManager;
+#endif
+
+
+// Class to track idle periods of time
+class IdleTimer {
+ public:
+  void Start()
+  {
+    is_idle_ = true;
+    start_time_ = std::chrono::steady_clock::now();
+  }
+
+  void Stop()
+  {
+    is_idle_ = false;
+    auto end = std::chrono::steady_clock::now();
+    auto duration = end - start_time_;
+    idle_ns_ += duration.count();
+  }
+
+  /// Restart the timer. This means that if the timer was already active, then
+  /// stop it (and count the pending time), and then restart it
+  ///
+  void Restart()
+  {
+    if (is_idle_) {
+      Stop();
+      Start();
+    }
+  }
+
+  /// Reset the time counter
+  ///
+  void Reset()
+  {
+    Restart();
```

Comment:
Shouldn't reset() just call stop() (if idle)? It's not intuitive to me that reset should also restart the timer.

Reply:
Restart is used at the end of a measuring region. It will stop and start (to account for the time up until the end).
Reset is used at the start of a new measuring region. It will stop and start (to make sure we are counting from the beginning of the region) and reset the counter to 0.